ovos_core/skill_installer.py

```diff
@@ -0,0 +1,174 @@
+import sys
+from os.path import join, exists
+from subprocess import Popen, PIPE
+from tempfile import gettempdir
+from typing import Optional
+
+from combo_lock import ComboLock
+from ovos_config.config import Configuration
+
+from ovos_bus_client import Message
+from ovos_utils.log import LOG
+from importlib import reload
+import ovos_plugin_manager
+
+
+class SkillsStore:
+    # default constraints to use if none are given
+    DEFAULT_CONSTRAINTS = '/etc/mycroft/constraints.txt'
+    PIP_LOCK = ComboLock(join(gettempdir(), "ovos_pip.lock"))
+
+    def __init__(self, bus, config=None):
+        self.config = config or Configuration().get("skills", {}).get("installer", {})
+        self.bus = bus
+        self.bus.on("ovos.skills.install", self.handle_install_skill)
+        self.bus.on("ovos.skills.uninstall", self.handle_uninstall_skill)
+        self.bus.on("ovos.pip.install", self.handle_install_python)
+        self.bus.on("ovos.pip.uninstall", self.handle_uninstall_python)
+
+    def shutdown(self):
+        pass
+
+    def play_error_sound(self):
+        self.bus.emit(Message("mycroft.audio.play_sound",
+                              {"uri": "snd/error.mp3"}))
+
+    def play_success_sound(self):
+        self.bus.emit(Message("mycroft.audio.play_sound",
+                              {"uri": "snd/acknowledge.mp3"}))
+
+    def pip_install(self, packages: list,
+                    constraints: Optional[str] = None,
+                    print_logs: bool = True):
+        if not len(packages):
+            self.play_error_sound()
+            return False
+        # Use constraints to limit the installed versions
+        if constraints and not exists(constraints):
+            LOG.error('Couldn\'t find the constraints file')
+            self.play_error_sound()
+            return False
+        elif exists(SkillsStore.DEFAULT_CONSTRAINTS):
+            constraints = SkillsStore.DEFAULT_CONSTRAINTS
+
+        pip_args = [sys.executable, '-m', 'pip', 'install']
+        if constraints:
+            pip_args += ['-c', constraints]
+        if self.config.get("break_system_packages", False):
+            pip_args += ["--break-system-packages"]
```

Comment:
Where does this configuration live, exactly? At the root of the config file or nested somewhere?

Reply:
```
"skills": {
    "installer": {
         "allow_pip": true,
         "break_system_packages": false
    }
}
```

Reply:
```
"skills": {
    "installer": {
         "allow_pip": true,
         "break_system_packages": false
    }
}
```